model/log.go

```diff
@@ -176,21 +176,21 @@ func GroupLines(lines []LogLine) ([]LogChunk, error) {
 	var currentChunk LogChunk
 
 	logChars := 0
-	for _, line := range lines {
-		if len(line.Msg) > maxLogChars {
+	for i := range lines {
+		if len(lines[i].Msg) > maxLogChars {
 			return nil, errors.New("Log line exceeded 4MB")
 		}
 
-		if len(line.Msg)+logChars > maxLogChars {
+		if len(lines[i].Msg)+logChars > maxLogChars {
 			logChars = 0
 			chunks = append(chunks, currentChunk)
 			currentChunk = LogChunk{}
 		}
 
-		logChars += len(line.Msg)
-		currentChunk.Lines = append(currentChunk.Lines, line)
-		if currentChunk.EarliestTime != nil && line.Time.Before(*currentChunk.EarliestTime) {
-			currentChunk.EarliestTime = &line.Time
+		logChars += len(lines[i].Msg)
+		currentChunk.Lines = append(currentChunk.Lines, lines[i])
+		if currentChunk.EarliestTime == nil || lines[i].Time.Before(*currentChunk.EarliestTime) {
+			currentChunk.EarliestTime = &lines[i].Time
```

Comment:
I think this is fine because Time objects are immutable, but it did cause me some alarm to create a pointer to that Time value.

Reply:
I removed the whole earliest time thing since it seems like we already rely on the assumption that log lines are already sorted by time.